### PR TITLE
Add 'Updating the boot loader on RHCOS nodes using bootupd' to update section & fix issues with the bootupd page

### LINKED ISSUE #71172: Create a dedicated page or section for bootupd documentation
### Which section(s) is the issue in?

We currently have sub-sections mentioning bootupd but it's included in the installation instructions (not in the updates section) and it's difficult to find or search for:
https://docs.openshift.com/container-platform/4.14/installing/installing_bare_metal/installing-bare-metal.html#architecture-rhcos-updating-bootloader.adoc_installing-bare-metal

### What needs fixing?

We should create a dedicated section in the documentation, likely close to how the updates are documented, for instructions on how to use bootupd.

### PR DESCRIPTION
**Pull request by travier**
Add 'Updating the boot loader on RHCOS nodes using bootupd' to update section

- Fix issues with the bootupd section: Correct terms, commands and
  details regarding how `bootupd` works.
- Use a Butane config to generate the MachineConfig.
- Moves the boot loader update content to a dedicated page in the update
  section and removes it from the installation pages.

Fixes: https://github.com/openshift/openshift-docs/issues/71172

---

Version(s):
All supported versions.

Issue:
https://issues.redhat.com/browse/OCPBUGS-28987

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

See also https://github.com/openshift/openshift-docs/issues/71172

### CODE REVIEW

**Comment by kalexand-rh** [comment]
@jldohmann, will you PTAL at this PR and help get the required reviews?

**Review by cgwalters — APPROVED** [review]

**Comment by cgwalters** [comment]
LGTM thanks!

**Comment by jldohmann** [comment]
preview: https://71174--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal#architecture-rhcos-updating-bootloader.adoc_installing-bare-metal

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 1]
```diff
@@ -10,17 +10,18 @@
 [id="architecture-rhcos-updating-bootloader.adoc_{context}"]
 = Updating the bootloader using bootupd
 
-To update the bootloader by using `bootupd`, you must either install `bootupd` on {op-system} machines manually or provide a machine config with the enabled `systemd` unit. Unlike `grubby` or other bootloader tools, `bootupd` does not manage kernel space configuration such as passing kernel arguments.
+To update the bootloader by using `bootupd`, you must either run the `bootupctl update` command on {op-system} machines manually or provide a machine config with a `systemd` unit.
 
-After you have installed `bootupd`, you can manage it remotely from the {product-title} cluster.
+Unlike `grubby` or other bootloader tools, `bootupd` does not manage kernel space configuration such as passing kernel arguments.
+To configure kernel arguments, see xref:../../nodes/nodes/nodes-nodes-managing.adoc#nodes-nodes-kernel-arguments_nodes-nodes-managing[Adding kernel arguments to Nodes].
```
Comment (jldohmann) [inline modules/architecture-rhcos-updating-bootloader.adoc:16]:
```suggestion
To configure kernel arguments, see xref:../../nodes/nodes/nodes-nodes-managing.adoc#nodes-nodes-kernel-arguments_nodes-nodes-managing[Adding kernel arguments to nodes].
```

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 2]
```diff
@@ -10,17 +10,18 @@
 [id="architecture-rhcos-updating-bootloader.adoc_{context}"]
 = Updating the bootloader using bootupd
 
-To update the bootloader by using `bootupd`, you must either install `bootupd` on {op-system} machines manually or provide a machine config with the enabled `systemd` unit. Unlike `grubby` or other bootloader tools, `bootupd` does not manage kernel space configuration such as passing kernel arguments.
+To update the bootloader by using `bootupd`, you must either run the `bootupctl update` command on {op-system} machines manually or provide a machine config with a `systemd` unit.
```
Comment (jldohmann) [inline modules/architecture-rhcos-updating-bootloader.adoc:13]:
```suggestion
To update the boot loader by using `bootupd`, you must either run the `bootupctl update` command on {op-system} machines manually or provide a machine config with a `systemd` unit.
```

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 3]
```diff
@@ -10,17 +10,18 @@
...
 
-After you have installed `bootupd`, you can manage it remotely from the {product-title} cluster.
+Unlike `grubby` or other bootloader tools, `bootupd` does not manage kernel space configuration such as passing kernel arguments.
+To configure kernel arguments, see xref:../../nodes/nodes/nodes-nodes-managing.adoc#nodes-nodes-kernel-arguments_nodes-nodes-managing[Adding kernel arguments to Nodes].
 
 [NOTE]
 ====
-It is recommended that you use `bootupd` only on bare metal or virtualized hypervisor installations, such as for protection against the BootHole vulnerability.
+You can use `bootupd` to update the bootloader to protect against the BootHole vulnerability.
```
Comment (jldohmann) [inline modules/architecture-rhcos-updating-bootloader.adoc:20]:
```suggestion
You can use `bootupd` to update the boot loader to protect against the BootHole vulnerability.
```

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 4]
```diff
@@ -10,17 +10,18 @@
...
 ====
 
-.Manual install method
-You can manually install `bootupd` by using the `bootctl` command-line tool.
+.Manual update method
+You can manually inspect the status of the system and update the bootloader by using the `bootupctl` command-line tool.
```
Comment (jldohmann) [inline modules/architecture-rhcos-updating-bootloader.adoc:24]:
```suggestion
You can manually inspect the status of the system and update the boot loader by using the `bootupctl` command-line tool.
```

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 5]
```diff
@@ -77,15 +78,15 @@ Updated: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
 ----
 
 .Machine config method
-Another way to enable `bootupd` is by providing a machine config.
+Another way to automatically update the bootloader during the boot process using `bootupd` is by providing a machine config.
```
Comment (jldohmann) [inline modules/architecture-rhcos-updating-bootloader.adoc:81]:
```suggestion
Another way to automatically update the boot loader during the boot process using `bootupd` is by providing a machine config.
```

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 6]
```diff
@@ -77,15 +78,15 @@ Updated: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
...
 
-* Provide a machine config file with the enabled `systemd` unit, as shown in the following example:
+* Provide a machine config file with the `systemd` unit, as shown in the following example:
 +
 .Example output
 [source,yaml]
```
Comment (jldohmann) [inline modules/architecture-rhcos-updating-bootloader.adoc:86]: the attribute in the code block isn't rendering, you need the substitution:
```suggestion
[source,yaml,subs="attributes+"]
```

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 7]
```diff
@@ -101,3 +102,8 @@ Another way to enable `bootupd` is by providing a machine config.
           [Install]
           WantedBy=multi-user.target
 ----
+
+[NOTE]
+====
+This `systemd` unit will try to update the bootloader on {op-system} machines every boot. Interrupting this process might lead to unbootable nodes.
```
Comment (jldohmann) [inline modules/architecture-rhcos-updating-bootloader.adoc:108]:
```suggestion
This `systemd` unit will try to update the boot loader on {op-system} machines every boot. Interrupting this process might lead to unbootable nodes.
```

**Review by jldohmann — COMMENTED** [review]
mostly style guide "boot loader" nits, with one comment on attribute substitution. otherwise looks good! thank you 😄

**Comment by jldohmann** [comment]
@mike-nguyen can you PTAL for QE ack? thank you

**Comment by travier** [comment]
Made the suggested fixes. Thanks

File: modules/architecture-rhcos-updating-bootloader.adoc [hunk 8]
```diff
@@ -77,15 +78,15 @@ Updated: grub2-efi-x64-1:2.04-31.fc33.x86_64,shim-x64-15-8.x86_64
...
-* Provide a machine config file with the enabled `systemd` unit, as shown in the following example:
+* Provide a machine config file with the `systemd` unit, as shown in the following example:
 +
 .Example output
-[source,yaml]
+[source,yaml,subs="attributes+"]
 ----
   variant: rhcos
-  version: 1.1.0
+  version: v{MaistraVersion}
```
Comment (travier) [inline modules/architecture-rhcos-updating-bootloader.adoc:89]: This has been replaced by `v2.4` and should be `v4.15.0` instead so this is probably not the right attribute.

Comment (jldohmann) [reply modules/architecture-rhcos-updating-bootloader.adoc:89]: if this is the RHCOS version then `{product-version}` ?

**Comment by mike-nguyen** [comment]
The content in general looks good to me.  We just need resolution on line 89.

File: modules/architecture-rhcos-updating-bootloader.adoc (hunk 8 above)
Comment (travier) [reply modules/architecture-rhcos-updating-bootloader.adoc:89]: It's the Butane spec version. I'll have to dig into other Butane example to see what we use.

Comment (travier) [reply modules/architecture-rhcos-updating-bootloader.adoc:89]: Hum, my bad, it's not a Butane config but directly a MachineConfig one. Will have to look again.

**Comment by travier** [comment]
OK, I've reworked the last section to follow the Butane path like we do in the other docs. Not tested yet.

**Comment by travier** [comment]
And I've moved the page to the update section: https://github.com/openshift/openshift-docs/issues/71172

**Comment by travier** [comment]
The new doc page is now at: https://71174--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/updating_a_cluster/updating-bootloader-rhcos

**Comment by jldohmann** [comment]
@travier can you squash your commits? thank you!

**Comment by travier** [comment]
Sure. Squashed and updated to only include bootupd changes. I'll make another PR for the other minor changes.

**Comment by jldohmann** [comment]
since 4.11 is EOL, i'm removing that branch label

**Comment by jldohmann** [comment]
@travier it looks like the auto cherrypicks to the 4.12-4.14 enterprise branches failed, and will need a manual cherrypick. is this something you can? if not, i can take a look later. thankfully the 4.15 auto cherrypick didn't fail 😄

**Comment by travier** [comment]
I'll take a look at them.

**Comment by travier** [comment]
- 4.14: https://github.com/openshift/openshift-docs/pull/71690
- 4.13: https://github.com/openshift/openshift-docs/pull/71689
- 4.12: https://github.com/openshift/openshift-docs/pull/71691

**Pull request merged**